### DISCUSSION TITLE
Dependency rule tried to blank-out primary key column 'serialized_dag.dag_id'

### DISCUSSION THREAD

**Discussion by obautista** [apache/airflow#29755] [Q&A]
### Apache Airflow version

2.5.1

### What happened

Just completed a fresh Airflow:
- Airflow: v2.5.1
- Ubuntu: v22.04
- MySql:  Ver 8.0.32-0ubuntu0.22.04.2 for Linux on x86_64 ((Ubuntu))

I am setting up maintenance tasks using https://github.com/teamclairvoyant/airflow-maintenance-dags

Airflow is reporting "Dependency rule tried to blank-out primary key column 'serialized_dag.dag_id'" when running Clear Missing Dags (https://github.com/teamclairvoyant/airflow-maintenance-dags/blob/master/clear-missing-dags/airflow-clear-missing-dags.py).

Is there a way to resolve this?
 

### How to reproduce

I am setting up maintenance tasks using https://github.com/teamclairvoyant/airflow-maintenance-dags

Airflow is reporting "Dependency rule tried to blank-out primary key column 'serialized_dag.dag_id'" when running Clear Missing Dags (https://github.com/teamclairvoyant/airflow-maintenance-dags/blob/master/clear-missing-dags/airflow-clear-missing-dags.py).

### Operating System

Ubuntu v22.04

### Versions of Apache Airflow Providers

v2.5.1

### Deployment

Other

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by hussein-awala** [comment]
This error is not related to Airflow, for that I convert the issue to discussion.

You are using code that has not been updated since August 2021, while we have made a lot of changes to Airflow DB and its schema since that date.

I suggest using the internal API to delete the dags, for example using the method [delete_dag](https://github.com/apache/airflow/blob/671b88eb3423e86bb331eaf7829659080cbd184e/airflow/api/common/delete_dag.py#L37-L37). And if you want to delete the dags in batch query, you can override the method by replacing `models.TaskInstance.dag_id == dag_id` by `models.TaskInstance.dag_id.in_(list_dag_ids)`
